test/src/cross_currencies.test.js

```diff
@@ -6,77 +6,36 @@ import Xrp from "@ledgerhq/hw-app-xrp";
-    XTZConfig, XTZConfigSignature
+    ETH_INFO,
+    AE_INFO,
+    XRP_INFO,
+    XLM_INFO,
+    XTZ_INFO,
+    LTC_INFO,
+    LTC_LEGACY_INFO,
+    BTC_INFO,
+    BTC_LEGACY_INFO,
+    BTC_NEWPROTOCOL_SEGWIT_INFO,
+    BTC_NEWPROTOCOL_TAPROOT_INFO,
 } from "./common";
-import Exchange from "./exchange.js";
-import {
-    TRANSACTION_RATES,
-    TRANSACTION_TYPES
-} from "./exchange.js";
 
 import { AppClient as BtcAppAclient, PsbtV2, DefaultWalletPolicy } from "./AppBtc"
 
-const TEZOS_ADDRESS_1 = "tz1RVYaHiobUKXMfJ47F7Rjxx5tu3LC35WSA";
-const TEZOS_ADDRESS_2 = "tz1RjJLvt7iguJQnVVWYca2AHDpHYmPJYz4d";
-const TEZOS_DERIVATION_PATH_1 = "44'/1729'/0'/0'";
+import { zemu, nano_environments } from './test.fixture';
 
-import Zemu from "@zondax/zemu";
-import { waitForAppScreen, zemu, nano_environments } from './test.fixture';
+import { SwapTransactionPerformer } from "./SwapTransactionPerformer"
 
 test('[Nano S] BTC (legacy protocol) swap to LTC fails', zemu(nano_environments[0], async (sim) => {
-    const swap = new Exchange(sim.getTransport(), TRANSACTION_TYPES.SWAP);
-    const transactionId: string = await swap.startNewTransaction();
-    await swap.setPartnerKey(partnerSerializedNameAndPubKey);
-    await swap.checkPartner(DERSignatureOfPartnerNameAndPublicKey);
-    var tr = new proto.ledger_swap.NewTransactionResponse();
-    tr.setPayinAddress("34dZAvAf1ywuKj1iAydSpPtavigteo1T5G");
-    tr.setPayinExtraId("");
-    tr.setRefundAddress("bc1qwpgezdcy7g6khsald7cww42lva5g5dmasn6y2z");
-    tr.setRefundExtraId("");
-    tr.setPayoutAddress("LKtSt6xfsmJMkPT8YyViAsDeRh7k8UfNjD");
-    tr.setPayoutExtraId("");
-    tr.setCurrencyFrom("BTC");
-    tr.setCurrencyTo("LTC");
-    tr.setAmountToProvider(numberToBigEndianBuffer(500000));
-    tr.setAmountToWallet(numberToBigEndianBuffer(10000000));
-    tr.setDeviceTransactionId(transactionId);
-
-
-    const payload: Buffer = Buffer.from(tr.serializeBinary());
-    await swap.processTransaction(payload, 1070);
-    const digest: Buffer = Buffer.from(sha256.sha256.array(payload));
-    const signature: Buffer = secp256k1.signatureExport(secp256k1.sign(digest, swapTestPrivateKey).signature);
-    await swap.checkTransactionSignature(signature);
-    const ltcAddressParams = getSerializedAddressParametersBTC("49'/0'/0'/0/0");
-    await swap.checkPayoutAddress(LTCConfig, LTCConfigSignature, ltcAddressParams.addressParameters);
-
-    const btcAddressParams = getSerializedAddressParametersBTC("84'/0'/0'/1/0", "bech32");
-    const checkRequest = swap.checkRefundAddress(BTCConfig, BTCConfigSignature, btcAddressParams.addressParameters);
-
-    // Wait until we are not in the main menu
-    await waitForAppScreen(sim);
-    await sim.navigateAndCompareSnapshots('.', 'nanos_btc_to_ltc_swap', [4, 0]);
-    await expect(checkRequest).resolves.toBe(undefined);
-
-    await swap.signCoinTransaction();
-
-    await Zemu.sleep(1000);
+    var test = new SwapTransactionPerformer(nano_environments[0], sim);
```

Comment:
Nits:
- `const` (better to forbid `var` completely imho; always `const`, or `let` if `const` doesn't work)
- the name `test` as a variable shadows the call to the `test` function from jest; maybe just `t` so that it's short? General I prefer descriptive name, but here minimizing the width of the rows might make sense.

The same comment applies to the other tests in the file, and in `tezos.test.js`.